### ISSUE TITLE
Customizing SVG Icon doesn't seem to work

### ISSUE THREAD

**Issue by MTD-Thomas** [lolochristen/OpenLayers.Blazor#96]
Hello,

I'm trying to make a layer with SVG Icons and customize them but it doesn't seem to work properly

I can display my svg icon like this (take care it's white by default so not very visible)
```csharp
string svg =
    "<svg width=\"16\" height=\"16\" viewBox=\"0 0 16 16\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\"><path d=\"M14.2222 8.77778H1.77778C1.35 8.77778 1 9.12778 1 9.55556V14.2222C1 14.65 1.35 15 1.77778 15H14.2222C14.65 15 15 14.65 15 14.2222V9.55556C15 9.12778 14.65 8.77778 14.2222 8.77778ZM4.11111 13.4444C3.25556 13.4444 2.55556 12.7444 2.55556 11.8889C2.55556 11.0333 3.25556 10.3333 4.11111 10.3333C4.96667 10.3333 5.66667 11.0333 5.66667 11.8889C5.66667 12.7444 4.96667 13.4444 4.11111 13.4444ZM14.2222 1H1.77778C1.35 1 1 1.35 1 1.77778V6.44444C1 6.87222 1.35 7.22222 1.77778 7.22222H14.2222C14.65 7.22222 15 6.87222 15 6.44444V1.77778C15 1.35 14.65 1 14.2222 1ZM4.11111 5.66667C3.25556 5.66667 2.55556 4.96667 2.55556 4.11111C2.55556 3.25556 3.25556 2.55556 4.11111 2.55556C4.96667 2.55556 5.66667 3.25556 5.66667 4.11111C5.66667 4.96667 4.96667 5.66667 4.11111 5.66667Z\" fill=\"white\"/></svg>";
svg = "data:image/svg+xml;utf8, " + svg;
Marker marker = new Marker(coordinate, svg, 16, 16, 8, 16);

// trying to change the color of the svg icon
// marker.Fill = "red";

_layer.ShapesList.Add(marker);
```

But modifying the color, size, or others properties don't do much, only `Scale` property works correctly.

Did I miss something ?

Can you help me on this please.

I also tried with a svg file and this is the same behavior

Thank you

**Comment by MTD-Thomas** [comment]
I just found that changing the color inside the style of icon works
```csharp
marker.Styles[0].Icon.Color = "red";
```

But I still can't figure out how to change the size.

**Comment by MTD-Thomas** [comment]
As the scale works, I can change the size with this parameters for now.

So what I didn't understand is that I need to change the `IconStyleOptions` inside the `Styles` of a `Shape` to define properties of my icon, I can't use other properties, is that right ?

**Comment by MTD-Thomas** [comment]
Here are my findings on svg icon.

I switch to svg file because it's way more easy to manage across all problems found.

- `Width`/`Height` on style icon doesn't work (it resize the container but not the image)
- `Scale` works but if used with `Color` I think it's not an svg anymore so the resize can be very pixelized (I had a 16x16 with scale 4 and it was all blurry)
- Working on svg data directly (no svg file) is a no go, because it doesn't understand color in hexadecimal only string color like `red`, `black` but not all of them either

**Comment by frankvandenbergh** [comment]
Hi Thomas,

This seems to work:
```
`protected override async Task OnAfterRenderAsync(bool firstRender)
{
    if (firstRender)
    {
        int radius = 13;
        string fillColor = "orange";
        string strokeColor = "black";
        string color = "red";
        int strokeWidth = 1;
        string text = "99";
        string fontSize = "0.7em";

        var svg = $"<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"{radius * 2}\" height=\"{radius * 2}\">" +
          $"<circle cx=\"{radius}\" cy=\"{radius}\" r=\"{radius - strokeWidth}\" stroke=\"{strokeColor}\" stroke-width=\"{strokeWidth}\" fill=\"{fillColor}\"/> " +
          $"<text x=\"50%\" y=\"50%\" text-anchor=\"middle\" font-size=\"{fontSize}\" dy=\".3em\" style=\"fill: {color}; \">{text}</text> " +
          "</svg>";
        svg = "data:image/svg+xml;utf8, " + svg;

          Marker marker = new Marker(new Coordinate(long, lat), svg, radius * 2, radius*2, radius, radius);
      //  marker.Scale = 1;

        _layer.ShapesList.Add(marker);
    }
}`
```
But the size is independent on the zoom level.

**Comment by MTD-Thomas** [comment]
Yes i tried like this too and indeed changing the width/height works like this, The width/height doesn't work with a svg file, that have a fixed width/height, then the width/height of the Style will not change it.

I stopped using this system because it's quite limited. For example if you try to use a color in hexadecimal (like any value with `#`) it will not work it puts black by default and some named color didn't work too (like yellow)

So overall, right now it's easier for me to use svg file but I will probably need to programmatically make it dynamic so I will check it again later when I have more time. But I don't know if it's a limitation of OpenLayers or your system because you seem to have linked everything correctly to OpenLayers so I don't see why it doesn't work

Thanks you for your answer

**Comment by frankvandenbergh** [comment]
I use `return $"rgb({color.R},{color.G},{color.B})";` for the coloring. Size seems to work with OSM:
![osm](https://github.com/user-attachments/assets/72a63ed1-5327-4f2a-88e7-115e2d4055e1)

**Comment by MTD-Thomas** [comment]
I tried with direct svg in string using color in `rgb([...])` format and it seems to work fine indeed

But I can't even make it work again like I used to: it doesn't update the shape now... So I probably mess up something

So I will retry it in a good way when I have time, maybe it was only the syntax of the color that cause my problem.

I will update this thread then with a good sample
